README.md

```diff
@@ -1,129 +1,201 @@
+
+`useStage` is a bool to either use current stage in beginning of path or not. The Default is `false`. For example, if you use it enabled (`true`) and your stage is `dev` the swagger will be in `dev/swagger`
+
+## Adding more details
+
+The default swagger file from vanilla Serverless framework will have the correct paths and methods but no details about the requests or responses.
+
+### API Summary and Details
+
+The optional attributes `summary` and `description` can be used to describe each HTTP request in Swagger.
+
+`swaggerTags` is an optional array that can be used to group HTTP requests with a collapsible name
+(i.e. grouping two endpoints `GET /dogs` and `POST /dogs` together).
+If not specified, all HTTP requests will be grouped under `default`.
+
+```js
+http: {
+    summary: 'This is a cool API',
+    description: 'Cool API description is here',
+    swaggerTags: ['Dogs']
+}
+```
+
+### Adding Data Types
+
+This plugin uses typescript types to generate the data types for the endpoints. By default, it pulls the types from `src/types/api-types.d.ts`.
+
+You can then assign these typescript definitions to requests as `bodyType` on the http or https config, or to the response as seen just below.
+
+### Responses
+
+You can also add expected responses to each of the http endpoint events. This is an object that contains the response code with some example details:
+
+```js
+responseData: {
+    // response with description and response body
+    200: {
+        description: 'this went well',
+        bodyType: 'helloPostResponse',
+    },
+
+    // response with just a description
+    400: {
+        description: 'failed Post',
+    },
+    // shorthand for just a description
+    502: 'server error',
+}
+```
+
+### Post request expected body
+
+When you create a `POST` or `PUT` endpoint, you expect to receive a specific structure of data as the body of the request.
+
+You can do that by adding a `bodyType` to the http event:
+
+```
+http: {
+    path: 'hello',
+    method: 'post',
+    cors: true,
+    bodyType: 'helloPostBody',
+}
+```
+
+### Query String Parameters
+
+If you want to specify the query string parameters on an endpoint you can do this by adding an object of `queryStringParameters` to the event (original I know). This has two required properties of `required` and `type` as well as an optional `description`.
+
+```
+http: {
+    path: 'goodbye',
+    method: 'get',
+    queryStringParameters: {
+        bob: {
+            required: true,
+            type: 'string',
+            description: 'bob',
+        },
+        count: {
+            required: false,
+            type: 'integer',
+        },
+    },
+},
+```
+
+![Query String Parameters](./doc_images/queryStringParams.png)
+
+### Multi-Valued Query String Parameters
+
+If you use multi-value query string parameters (array), then you must specify that your `type` is `array` and specify your data type (string or integer) in `itemsType`

```

Comment:
I think this is my last comment to address other than those unit tests, sorry I missed this in my review a minute ago
```suggestion
If you use multi-value query string parameters (array), then you must specify that your `type` is `array` and specify your data type (string or integer) in `arrayItemsType`

```